### DISCUSSION TITLE
field overiding failed

### DISCUSSION THREAD

**Discussion by Johnssonslotion** [pydantic/pydantic#6875] [Question] [answered]
```
class CommonRequest(BaseModel):
url:str
params: CommonParams ### BaseModel class
....

class SpecificRequest(CommonRequest):
params: SpecificParams ### CommonParams with url property (str)
@computed_field
@property
def url(self)->str:
   return self.params.url  ###  url_str_1 

@computed_field
@property
def urls(self)->str:
  return "test"

```

whatever 'url_str_1' values set,
SepcificRequest obj model_dump -> url:<property object at memory_address> 
but obj.urls is normally work (urls:test)

**Comment by braindevices** [comment]
apparently you do not need subclass to accomplish what you want. Please provide the full executable example to let us help you.

Since your url and params are completely different replace `class SpecificRequest(CommonRequest):` to `class SpecificRequest(BaseModel)` make it works.

**Comment by braindevices** [comment]
I don't thing you are supposed to override the normal field with computed field at all.

I think normally they do not support override field definition at all. you only add more fields.

If you want to do this, you probably should create more intermedia base model as suggested here:
https://github.com/pydantic/pydantic/discussions/2686#discussioncomment-631025

The entire override existing class member with a property entry is problematic on pure python level already.
Consider following code:
```
class A:
    x: int
    y: int
    def __init__(self, x, y) -> None:
        self.x = x
        self.y = y

class B(A):
    def __init__(self, y) -> None:
        super().__init__(y*2, y)


class C(A):
    def __init__(self, y) -> None:
        super().__init__(0, y)

    @property
    def x(self):
        return self.y*2


A(1,2)
B(2)
C(3)
```

Apparently this won't work at all. Because by definition the attribute x in C won't be settable at all.

One will simply set the x during `__init__`.

**Comment by braindevices** [comment, marked as answer]
some possible alternative: always use computed field:

```
from typing import Any
from pydantic import BaseModel, computed_field, PrivateAttr


class Shape(BaseModel):
    _area: float = PrivateAttr()

    def __init__(self, **data: Any) -> None:
        super().__init__(**data)
        self._area = data.get("area", 0.0)

    @computed_field
    def area(self) -> float:  # converted to a `property` by `computed_field`
        return self._area

class Square(Shape):
    width: float
    @computed_field
    def area(self) -> float:  # converted to a `property` by `computed_field`
        return self.width**2

print(Shape(area=3.0).model_dump())
print(Shape.model_validate({'area': 3.0}))
print(Shape.model_validate({'area': "hello"}))
print(Square(width=2).model_dump())
print(Square.model_validate({'width': 2}))
```

But you can see the `print(Shape.model_validate({'area': "hello"}))` works, so this is a problem. I think you need to use validator to automatically generate the `_area` here instead of directly override the `__init__`

**Reply by braindevices** [reply]
puah, I did not expect this will be the accepted answer though. The area in Shape model cannot be validated, unless you explicitly implement the validation by yourself in `__init__`. Also the type annotation on the privateattr do nothing actually.

I highly suggest creating more intermedia base models instead of this hacky way.
